compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -640,6 +640,17 @@ impl<'a> Parser<'a> {
             }
         }
 
+        // extra info for `c"str"` before 2021 edition or `c "str"` in all editions
+        if self.prev_token.is_ident_named(Symbol::intern("c"))
+            && let TokenKind::Literal(token::Lit { kind: token::Str | token::StrRaw(..), .. }) =
+                &self.token.kind
+        {
+            err.note("you may be trying to declare a CStr literal");
```

Comment:
You could think about making this diagnostic translatable. https://rustc-dev-guide.rust-lang.org/diagnostics/translation.html#writing-translatable-diagnostics